docs/scripting/language/Tags.md

```diff
@@ -0,0 +1,170 @@
+
+## `Operators`
+Operators such as +, ==, > etc can be overloaded for different tags, i.e. doing + on two Float:s does something different to doing it on two non-tagged variables. This is especially useful in the case of float variables as as mentioned they are not really floats they are integers with a very specific bit pattern, if the operators were not overloaded the operations would simply be performed on the integers which would give gibberish if the answer were interpreted as a float again. For this reason the Float: tag has overloaded versions of most of the operators to call special functions to do the maths in the server instead of in pawn.
+
+An operator is exactly like a normal function but instead of a function name you use "operator(**symbol**)" where (**symbol**) is the operator you want to overwrite. The valid operators are:
+```c
++
+-
+=
+++
+--
+==
+*
+/
+!=
+>
+<
+>=
+<=
+!
+%
+```
+
+Things like *= etc are done automatically. Things like & etc can't be overloaded. You can also overload an operator multiple times with different combinations of tag. For example:
+```c
+stock Float:operator=(Mytag:oper)
+{
+    return float(_:oper);
+}
+```
+
+If you add that to your code and do:
+```c
+new
+    Float:myFloat,
+    Mytag:myTag;
+ 
+myFloat = myTag;
+```
+
+You will no longer get a compiler warning as you would have before because the '=' operator for the case Float: = Mytag: is now handled so the compiler knows exactly what to do.
+
+## `Overwriting`
+In the overloading example above the functional line was:
+```c
+return float(_:oper);
+```
+
+This is an example of tag overwriting, the "_:" in front of oper means the compiler basically ignores the fact that oper has a tag type of Mytag: and treats it as tag type _: (i.e. no tag type). The function float() tags a normal number so must be passed one. In that example it is assumed that Mytag stores an ordinary integer but overwriting must be dealt with very carefully, for example the following will give very odd results:
+```c
+new
+    Float:f1,
+    Float:f2 = 4.0;
+f1 = float(_:f2);
+```
+
+Sense would dictate that f1 would end up as 4.0, however it won't. As mentioned f2 stores a representation of 4.0, not just 4 as an integer would, this means the actual value of the variable as an integer is a very odd number. Thus if you tell the compiler to treat the variable as an integer it will simply take the bit pattern in the variable as the value, it won't convert the float to an integer, so you will get an almost random number (it's not actually random as there's a pattern to IEEE floating points but it will be nothing like 4.0).
+
+If you know what you're doing this can be used to suppress warnings but you should investigate whether it's exactly what's wanted first and that the warning isn't actually telling you something important.
+
+## `Enums`
```

Comment:
"Draft pull request" is a neat github feature for marking work as in-progress!

Reply:
removed.